backend/src/v5/processors/users.js

```diff
@@ -72,4 +73,16 @@ Users.getAvatar = getAvatar;
 
 Users.uploadAvatar = uploadAvatar;
 
+Users.resetPasswordToken = (username) => resetPasswordToken(username);
+
+Users.resetPassword = async (username, token, newPassword) => {
+	const { customData } = await getUserByUsername(username);
```

Comment:
Few things here:
1. test the user validity/token validity at middleware So at the processor you're just performing the operation instead of testing preconditions.
2. you want to throw invalid token even if username doesn't exist.
3. Let the database query check the token validity for you by passing in the token/date you expect - so you don't have to do L81 manually.